report.py

```diff
@@ -56,34 +35,79 @@ def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}):
   "Authorization": f"Bearer {ENV['ZOOM_TOKEN']}"
 }
 
+ZOOM_BASE_URL = ENV.get('ZOOM_BASE_URL', "")
+DEFAULT_SLEEP_TIME = ENV.get('DEFAULT_SLEEP_TIME', "10")
 
-def run_report(param_attribute, headers, json_attribute_name, page_token=False):
 
-    url = ENV[f'ZOOM_{param_attribute}_URL']
-    params = ENV[f'ZOOM_{param_attribute}_PARAMS']
+# Functions
+def get_total_page_count(url: str, headers: Dict[str, Union[str, int]] = {}, params: dict = {}):
+    # get the total page count
+    total_page_count = 0
+    response = requests.request("GET", f"{url}", headers=headers, params=params)
+    status_code = response.status_code
+    if status_code != 200:
+        logger.warning(f'Received irregular status code: {status_code}')
+        logger.info('No page at all!')
+    else:
+        try:
+            results = json.loads(response.text.encode('utf8'))
+            total_page_count = results['page_count']
+        except json.JSONDecodeError:
+            logger.warning('JSONDecodeError encountered')
+            logger.info('No page at all!')
+    return total_page_count
 
-    # construct param string
-    f = furl.furl('')
-    f.args = params
-    print(f.url)
 
-    # get total page count
-    total_page_count = get_total_page_count(f"{url}{f.url}", headers)
+def run_report(api_url: str, headers: dict, json_attribute_name: str,
+               default_params: dict = {}, page_token: bool = False, use_date: bool = False):
+    url = ZOOM_BASE_URL+api_url
+    params = default_params
+    params['page_size'] = 300
```

Comment:
can use fixed page_size for all calls?

Reply:
Yeah, all of these were 300, I can switch this to a parameter though just incase.